src/opm/output/eclipse/Summary.cpp

```diff
@@ -204,10 +204,14 @@ namespace {
 
         constexpr Opm::EclIO::SummaryNode::Category category { Opm::EclIO::SummaryNode::Category::Segment };
         const std::vector<std::pair<std::string,Opm::EclIO::SummaryNode::Type>> requiredVectors {
-            { "SOFR", Opm::EclIO::SummaryNode::Type::Rate     },
-            { "SGFR", Opm::EclIO::SummaryNode::Type::Rate     },
-            { "SWFR", Opm::EclIO::SummaryNode::Type::Rate     },
-            { "SPR",  Opm::EclIO::SummaryNode::Type::Pressure },
+            { "SOFR",  Opm::EclIO::SummaryNode::Type::Rate     },
+            { "SGFR",  Opm::EclIO::SummaryNode::Type::Rate     },
+            { "SWFR",  Opm::EclIO::SummaryNode::Type::Rate     },
+            { "SPR",   Opm::EclIO::SummaryNode::Type::Pressure },
+            { "SPRD",  Opm::EclIO::SummaryNode::Type::Pressure },
+            { "SPRDH", Opm::EclIO::SummaryNode::Type::Pressure },
+            { "SPRDF", Opm::EclIO::SummaryNode::Type::Pressure },
+            { "SPRDA", Opm::EclIO::SummaryNode::Type::Pressure },
```

Comment:
Just a quick note: This list is for those summary vectors that are needed for restart and which should be reported in the `RSEG` array of the restart file.

Reply:
Okay, thanks.  Basically the task is to add `SPRD, SPRDH, SPRDF, SPRDA` to summary list. 

Could you please give some instructions on how I should proceed?  I think this is probably the most efficient way to do this. 

The downstream PR (probably WIP) is created. 
https://github.com/OPM/opm-simulators/pull/2531

Reply:
> Just a quick note: This list is for those summary vectors that are needed for restart and which should be reported in the RSEG array of the restart file.

Thanks. I do not think they are needed for restart. I see them as for output/analysis/debug purposes.

Reply:
> Could you please give some instructions on how I should proceed?

Sure.  You need to take the following steps

1. Ensure that the new vectors are recognized as [segment quantities](https://github.com/OPM/opm-common/blob/e980713572cc82b94b983baf83007d954e57b53a/src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp#L556) in the `SummaryConfig` class.

2. Ensure that they are recognized as [pressure](https://github.com/OPM/opm-common/blob/e980713572cc82b94b983baf83007d954e57b53a/src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp#L179) quantities.

3. Write evaluation functions similar to [spr](https://github.com/OPM/opm-common/blob/e980713572cc82b94b983baf83007d954e57b53a/src/opm/output/eclipse/Summary.cpp#L571-L593) that will pick up the pertinent values from the expanded `data::Segment` structure.  By the way, those should also be added to
     ```C++
     data::Segment::operator==(const Segment&) const
    ```

4. Hook those evaluation functions into the global [function table](https://github.com/OPM/opm-common/blob/e980713572cc82b94b983baf83007d954e57b53a/src/opm/output/eclipse/Summary.cpp#L1212) keyed on their summary keyword.

Reply:
> > Just a quick note: This list is for those summary vectors that are needed for restart and which should be reported in the `RSEG` array of the restart file.
> 
> Thanks. I do not think they are needed for restart. I see them as for output/analysis/debug purposes.

Cool.  In that case, please remove the quantities from the list.  Otherwise, they will be calculated and added to `SummaryState` for all segments of all multi-segmented wells at all times-

Reply:
For the point 2, 
https://github.com/OPM/opm-common/blob/e980713572cc82b94b983baf83007d954e57b53a/src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp#L179

The segment pressure `SPR` is not there. Is there something missing here?  Not sure whether `PR` is covering those? But `PR` is also used for many production rates. 

@bska

Reply:
> The segment pressure `SPR` is not there.

Actually it is, it's just a little sneaky.  The **argument** to the `is_in_set()` function is [`keyword.substr(1)`](https://github.com/OPM/opm-common/blob/e417ca6688841bedb90be6cf1b863ad708047c62/src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp#L184) which strips off the initial `S` character.

Reply:
Is the following for performance reason?

https://github.com/OPM/opm-common/blob/e980713572cc82b94b983baf83007d954e57b53a/src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp#L560

Reply:
> Is the following for performance reason?

Partly.  It quickly eliminates a large number of keywords that we haven't explicitly recognized as segment-related.